### PR TITLE
Issue 847 deprecated api usage 

### PR DESCRIPTION
**Pull request by aparnamichael**
Part of #847 
1. Replaced 'ExtensionPointName.extensions()' with 'getExtensionList().stream()
2. Replaced ServiceManager.getService with project.getService.
3. Replaced deprecated field ProjectTopics.MODULES with ModuleListener.TOPICS.
4. Updated deprecated Notification constructor and removed deprecated setListener method.

### CODE REVIEW

File: src/main/java/io/openliberty/tools/intellij/lsp4mp/MicroProfileDeploymentSupport.java [hunk 1]
```diff
@@ -38,7 +37,7 @@ public class MicroProfileDeploymentSupport implements ClasspathResourceChangedMa
     private final Project project;
 
     public static MicroProfileDeploymentSupport getInstance(Project project) {
-        return ServiceManager.getService(project, MicroProfileDeploymentSupport.class);
```
Comment (aparnamichael) [inline src/main/java/io/openliberty/tools/intellij/lsp4mp/MicroProfileDeploymentSupport.java:41]: Definition of  ServiceManager.getService() method.
  
  /*Deprecated
Use ComponentManager. getService(Class): Application. getService() / Project. getService(). */
  @Deprecated
  public static <T> T getService(@NotNull Project project, @NotNull Class<T> serviceClass) {
    return project.getService(serviceClass);
  }

File: src/main/java/io/openliberty/tools/intellij/lsp4mp/MicroProfileProjectService.java [hunk 2]
```diff
@@ -90,7 +89,7 @@ public MicroProfileProjectService(Project project) {
         LibraryTablesRegistrar.getInstance().getLibraryTable(project).addListener(this, project);
         connection = ApplicationManager.getApplication().getMessageBus().connect(project);
         connection.subscribe(VirtualFileManager.VFS_CHANGES, this);
-        project.getMessageBus().connect().subscribe(ProjectTopics.MODULES, this);
```
Comment (aparnamichael) [inline src/main/java/io/openliberty/tools/intellij/lsp4mp/MicroProfileProjectService.java:93]: /**
   * @deprecated use {@link ModuleListener#TOPIC} instead
   */
  @Deprecated 
  @Topic.ProjectLevel
  public static final Topic<ModuleListener> MODULES = ModuleListener.TOPIC;

File: src/main/java/io/openliberty/tools/intellij/lsp4mp4ij/psi/core/PropertiesManagerForJava.java [hunk 3]
```diff
@@ -192,7 +192,7 @@ private void collectDefinition(String uri, PsiFile typeRoot, PsiElement hyperlin
                     // Collect all adapted definition participant
                     JavaDefinitionContext context = new JavaDefinitionContext(uri, typeRoot, utils, module,
                             hyperlinkedElement, hyperlinkedPosition);
-                    List<IJavaDefinitionParticipant> definitions = IJavaDefinitionParticipant.EP_NAME.extensions()
```
Comment (aparnamichael) [inline src/main/java/io/openliberty/tools/intellij/lsp4mp4ij/psi/core/PropertiesManagerForJava.java:195]: @Deprecated("Use {@code getExtensionList().stream()}", ReplaceWith("getExtensionList().stream()"), DeprecationLevel.ERROR)
  fun extensions(): Stream<T> = getPointImpl(null).asSequence().asStream()

File: src/main/java/io/openliberty/tools/intellij/lsp4mp4ij/psi/core/PropertiesManagerForJava.java [hunk 4]
```diff
@@ -192,7 +192,7 @@ private void collectDefinition(String uri, PsiFile typeRoot, PsiElement hyperlin
...
+                    List<IJavaDefinitionParticipant> definitions = IJavaDefinitionParticipant.EP_NAME.getExtensionList().stream()
```
Comment (aparnamichael) [inline src/main/java/io/openliberty/tools/intellij/lsp4mp4ij/psi/core/PropertiesManagerForJava.java:195]: Definition of getExtensionList()
val extensionList: List<T>
    get() = getPointImpl(null).extensionList

File: src/main/java/io/openliberty/tools/intellij/actions/ViewIntegrationTestReport.java [hunk 5]
```diff
@@ -54,13 +53,12 @@ protected void executeLibertyAction(LibertyModule libertyModule) {
...
 
         if (failsafeReportVirtualFile == null || !failsafeReportVirtualFile.exists()) {
-            Notification notif = new Notification(Constants.LIBERTY_DEV_DASHBOARD_ID
-                    , LibertyPluginIcons.libertyIcon
-                    , LocalizedResourceUtil.getMessage("integration.test.report.does.not.exist.notification.title")
-                    , ""
-                    , LocalizedResourceUtil.getMessage("test.report.does.not.exist", failsafeReportFile.getAbsolutePath())
-                    , NotificationType.ERROR
-                    , NotificationListener.URL_OPENING_LISTENER);
+            Notification notif = new Notification(Constants.LIBERTY_DEV_DASHBOARD_ID,
```
Comment (vaisakhkannan) [inline src/main/java/io/openliberty/tools/intellij/actions/ViewIntegrationTestReport.java:56]: @aparnamichael , How many parameters can this method `Notification` actually have?
I can see the change in the number of parameters in other usages of same method `Notification ` , So Can you confirm , the changes are ok?

**Review by vaisakhkannan — CHANGES_REQUESTED** [review]

File: src/main/java/io/openliberty/tools/intellij/actions/LibertyGeneralAction.java [hunk 6]
```diff
@@ -157,11 +156,9 @@ protected final String[] toProjectNamesTooltips(@NotNull List<LibertyModule> lis
      * @param errMsg
      */
     protected void notifyError(String errMsg, Project project) {
-        Notification notif = new Notification(Constants.LIBERTY_DEV_DASHBOARD_ID, errMsg, NotificationType.WARNING)
-                .setTitle(LocalizedResourceUtil.getMessage("liberty.action.cannot.start"))
-                .setIcon(LibertyPluginIcons.libertyIcon)
-                .setSubtitle("")
-                .setListener(NotificationListener.URL_OPENING_LISTENER);
```
[the remark this answers is not in the document]
Comment (aparnamichael) [reply src/main/java/io/openliberty/tools/intellij/actions/LibertyGeneralAction.java:164]: There is no difference in the UI component, the change I made is removed the deprecated constructor and used the  recommended constructor.

```
@Deprecated 
public Notification(     @NotNull String groupId,
    @Nullable javax. swing. Icon icon,
    @Nullable String title,
    @Nullable String subtitle,
    @Nullable String content,
    @NotNull com. intellij. notification. NotificationType type,
    @Nullable com. intellij. notification. NotificationListener listener )
```
   
Newly used constructors


```
public Notification(     @NotNull String groupId,
    @NotNull String content,
    @NotNull NotificationType type )


public Notification(     @NotNull String groupId,
    @NotNull String title,
    @NotNull String content,
    @NotNull NotificationType type )
```
    
    
    and setIcon method
    `public @NotNull Notification setIcon(     @Nullable Icon icon )`
    
image

File: src/main/java/io/openliberty/tools/intellij/actions/ViewIntegrationTestReport.java (hunk 5 above)
Comment (aparnamichael) [reply src/main/java/io/openliberty/tools/intellij/actions/ViewIntegrationTestReport.java:56]: There are 2 recommended constructors, as I mentioned in the above comment. One with 3 parameter and another with 4 parameter. ie, the constructor with 3 parameter doesn't have an option to add title. We have title for all the notifications so I I only used the constructor with 4 parameters every where.

```
public Notification(     @NotNull String groupId,
    @NotNull String title,
    @NotNull String content,
    @NotNull NotificationType type )
```

File: src/main/java/io/openliberty/tools/intellij/actions/LibertyGeneralAction.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (vaisakhkannan) [reply src/main/java/io/openliberty/tools/intellij/actions/LibertyGeneralAction.java:164]: Thanks for the explanation

File: src/main/java/io/openliberty/tools/intellij/lsp4mp/MicroProfileDeploymentSupport.java (hunk 1 above)
Comment (mrglavas) [reply src/main/java/io/openliberty/tools/intellij/lsp4mp/MicroProfileDeploymentSupport.java:41]: Thanks for providing this here.

**Review by mrglavas — APPROVED** [review]
Looks good. Thanks for providing references to the recommended replacement APIs. It made it much easier to review this PR.

**Pull request merged**